proposals/helm.adoc

```diff
@@ -0,0 +1,122 @@
+
+*Current Core Maintainers*:
+
+* Adam Reese
+* Adnan Abdulhussein
+* Justin Scott
+* Maciej Kwiek
+* Matt Butcher
+* Matt Farina
+* Matt Fisher
+* Michelle Noorali
+* Nikhil Manchanda
+* Taylor Thomas
+* Vic Iglesias
+
+_Note, the current core maintainers represent 5 different companies._
+
+Sub-projects of Helm have their own maintainers. For example, you can read about the Charts maintainers in the link:https://github.com/kubernetes/charts/blob/master/OWNERS[OWNERS file].
+
+*Infrastructure requirements*: CI, CNCF Cluster, Object Storage
+
+*Issue tracker*: https://github.com/kubernetes/helm/issues
+
+Sub-projects each have their own issue queue.
+
+*Mailing lists*
+
+* Slack:
+** Helm Dev room https://kubernetes.slack.com/messages/helm-dev
+** Helm Users room https://kubernetes.slack.com/messages/helm-users (see https://kubernetes.slackarchive.io/helm-users/page-100)
+** Charts room https://kubernetes.slack.com/messages/charts
+** Chartmuseum room https://kubernetes.slack.com/messages/chartmuseum
+* https://lists.cncf.io/g/cncf-kubernetes-helm
+
+*Website*: http://helm.sh
+
+*Release methodology and mechanics*
+
+Helm uses link:http://semver.org/[semantic versioning] for releases. Releases are announced using GitHub releases while the release artifacts are placed into object storage for later download. The continuous integration systems, currently CircleCI, automatically places releases and development builds into object storage.
+
+Helm is currently releases stable releases with a major version of 2. When a minor version comes out containing new features a release branch is created where release candidates, final releases, and patch releases are created from. Anything to be added to these releases is cherry-picked into the branch prior to releases.
+
+The Helm release process is documented in the link:https://github.com/kubernetes/helm/blob/master/docs/release_checklist.md[release checklist].
+
+Sub-projects have their own releases processes. For example, the Helm Community Charts repository uses continuous deployments. All changes to individual charts increment the chart versions. A sync job runs every 15 minutes to pickup changes, builds the chart packages, and places them into object storage to be retrieved by Helm clients.
+
+*Social media accounts*:
+
+* https://twitter.com/helmpack
+* link:https://www.youtube.com/channel/UC_kvCKc5EHNomq64f8C4sfA[YouTube]
+
+*Existing sponsorship*:
+
+* Microsoft
+* Google
+* Codefresh
+* Bitnami
+* Ticketmaster
+* Codecentric
+
+_Note, these companies and their logos are listed on the link:https://helm.sh[Helm website]._
+
+*Adopters*:
```

Comment:
[CNCF Graduation Requirements](https://www.cncf.io/projects/graduation-criteria/) state:

"Document that it is being used successfully in production by at least three independent end users which, in the TOC’s judgement, are of adequate quality and scope."

Could you clarify which of the listed adopters are using Helm successfully in production, and provide some details of these uses.

Reply:
@quinton-hoole I updated the list to include a note about who at the conference shared how they were using Helm in production. Is this sufficient or would you like me to change up the list to be those using who've publicly stated they use it in production?

Reply:
Thanks Matt.  What we'd like to understand is some of the detail behind the actual production use cases, to help gauge the scope and quality.  For those of us who were not present at the conference, are there some slides or presentation videos that might include that info?  I looked around a bit, but couldn't find anything.

Reply:
@quinton-hoole There is a [YouTube playlist with the videos from the Helm Summit](https://www.youtube.com/playlist?list=PL69nYSiGNLP3PlhEKrGA0oN4eY8c4oaAH).

Another point, from the Kubernetes Application Survey, is that 26.7% of Helm users are using the community charts in production. This was a question we asked. Others use it as templates, or for demos, or just don't use them. We didn't collect data on the size or details of these production use cases.

Reply:
Thanks, I'll dig through the links Brian sent.  I was hoping for a summary, saying something like:

n major production use cases are as follows:

- Company A
   - Brief application description
   - Brief description of multiple environments it's deployed in, including size, complexity and update frequency.

- Company B
...

Reply:
@quinton-hoole I'm happy to go into more detail. Knowing the kind of thing you're looking for will help me make sure I hit the points you'd like to see. Thanks for providing that framework. I'm not sure I can hit on all your points for all of those using it but I'll try to do my best.

For these bullets I'll continue with the companies from the Helm Summit.

* Reddit
  * reddit.com which is in the process of moving to a SOA. The applications they are using helm for are, last I checked, stateless. It was stated that they aren't quite comfortable putting stateful services in Kubernetes, yet. They have not shared the exact microservices that are being deployed via Helm
  * Their environments include local development and amazon via multiple accounts. The environments are local dev, staging, and production. They keep building separate from production and use varying amazon controls including multiple-accounts and Kubernetes clusters in these accounts
  * Their CI is Drone and they use the drone-helm plugin. Charts are stored in S3
  * For config management they are using a project called [helmfile](https://github.com/roboll/helmfile) and the workflow is essentially gitops
  * While that have stated they use Helm in production they did not say which services, how high they scale, or how often they update. Helm is used at many stages of their workflow including local dev, staging, and production
* Ubisoft
  * Helm and Kubernetes is used for some of their games (did not list which ones).
  * Multiple clusters per production (on-premise, GKE, etc)
  * One of their goals is sharing and Helm is central to that. To share microservices between games
  * They have 60 charts with over 1,000 chart versions in their repository because they are doing continuous delivery
  * They have built tools on Helm to help automate some of their tasks
  * Helm is used as part of CI in GitLab CI
  * For configuration management they are using Terraform which has a plugin to work with Helm
* WP Engine
  * WP Engine provides hosted WordPress where they managing more than 300,000 wordpress sites for customers
  * As of the helm summit they had more than 7,000 releases of WordPress going through Helm. A release to wordpress site is not a 1:1 mapping
  * To manage all the configuration data about each of these sites they are using a project called [lostromos](https://github.com/wpengine/lostromos). Lostromos knows how to work with Helm
  * Jenkins is used for CI

Note, each of these is using Helm for package management and something else for config management. Each is using a different tool for this. In addition, each of these are using Helm in a different CI tool.

In addition to these there are other ways to look at production and users. We have [user profiles](https://github.com/kubernetes-helm/community/blob/master/user-profiles.md) to describe the different roles. One of those is Application Distributor and Bitnami is an excellent example of this who was doing application distribution prior to Kubernetes and still does for other platforms. They distribute numerous charts for common applications (e.g., mariadb, wordpress, and drupal).

If there is interest in more detail or other examples please let me know. Also, I'm not intimately familiar with these systems. If someone has a correction or more detail that information would be appreciated.

Reply:
Thanks @mattfarina .  That addresses my question well.